Foundation/DateIntervalFormatter.swift

```diff
@@ -28,8 +28,38 @@ open class DateIntervalFormatter : Formatter {
         NSUnimplemented()
     }
 
-    public required init?(coder: NSCoder) {
-        NSUnimplemented()
+    public required init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            preconditionFailure("Unkeyed coding is unsupported.")
+        }
+        
+        self.dateStyle = Style(rawValue: UInt(aDecoder.decodeInt64(forKey: "NS.dateStyle"))) ?? .noStyle
+        self.timeStyle = Style(rawValue: UInt(aDecoder.decodeInt64(forKey: "NS.timeStyle"))) ?? .noStyle
+        super.init(coder: aDecoder)
+        self.dateTemplate = aDecoder.decodeObject(of: NSString.self, forKey: "NS.dateTemplate") as String? ?? ""
+        /*self.dateTemplateFromStyles = aDecoder.decodeObject(of: NSString.self, forKey: "NS.dateTemplateFromStyles") as String?
```

Comment:
These keys are available in Darwin's Foundation. I'm not sure how they are used. I've put them in comments thus.

Reply:
Yes, I've preemptively implemented coding for some classes which are not implemented yet. Including`DateIntervalFormatter`, `DateComponentsFormatter`, `NSMeasurement`, etc.

Reply:
If we accept this, can we include a comment in the `NSUnimplemented()` methods of those classes pointing to the places we need to come back to uncomment?

Also: Rather than comment this, can you use `#if NS_IMPLEMENTS_DECODING_FOR_SR_9004`/`#endif` blocks instead? (https://bugs.swift.org/browse/SR-9004)

Reply:
@millenomi I didn’t know there is a SR for that. Tomorrow I will implement your suggestion. I will use that condition for few other similiar classes too.
This SR says Darwin compatibility for all classes, does CFNetwork classes should be compatible too?  I found they are not use same way of coding.

Reply:
I just opened it :) this is a good start, and the rest can be follow-ups, including the CFNetwork ones.

Reply:
Seems promising. I can work on CFNetwork classes in next PR which would implement unarchiving from files saved in Darwin for all classes.

Reply:
Seems promising. I can work on CFNetwork classes in next PR which would implement unarchiving from files saved in Darwin for all classes.

Reply:
Okay!